x-pack/plugins/canvas/public/state/reducers/elements.js

```diff
@@ -122,26 +123,19 @@ export const elementsReducer = handleActions(
         trimElement(element)
       );
     },
-    [actions.duplicateElement]: (workpadState, { payload: { pageId, element } }) => {
+    [actions.insertNodes]: (workpadState, { payload: { pageId, elements } }) => {
       const pageIndex = getPageIndexById(workpadState, pageId);
       if (pageIndex < 0) {
         return workpadState;
       }
-      return push(
-        workpadState,
-        ['pages', pageIndex, getLocation(element.position.type)],
-        trimElement(element)
-      );
-    },
-    [actions.rawDuplicateElement]: (workpadState, { payload: { pageId, element } }) => {
-      const pageIndex = getPageIndexById(workpadState, pageId);
-      if (pageIndex < 0) {
-        return workpadState;
-      }
-      return push(
-        workpadState,
-        ['pages', pageIndex, getLocation(element.position.type)],
-        trimElement(element)
+      return elements.reduce(
```

Comment:
nit: while I appreciate this bit, building an object and using [assign](https://www.npmjs.com/package/object-path-immutable#assign-initialobject-path-payload) might be easier to comprehend.